### PR TITLE
fix: added support for arrays in only and except

### PR DESCRIPTION
**Pull request by vkalta**

### CODE REVIEW

File: src/lib/entry-queryable.ts [hunk 1]
```diff
@@ -15,8 +15,16 @@ export class EntryQueryable extends BaseQuery {
    * @param {string} fieldUid - field uid to select
    * @returns {EntryQueryable} - returns EntryQueryable object for chaining method calls
    */
-  only(fieldUid: string): EntryQueryable {
-    this._queryParams['only[BASE][]'] = fieldUid;
+  only(fieldUid: string|string[]): EntryQueryable {
```
Comment (sunil-lakshman) [inline src/lib/entry-queryable.ts:18]: @vkalta Can you add the testcases for this implementation?

File: src/lib/entry-queryable.ts [hunk 2]
```diff
@@ -34,8 +42,16 @@ export class EntryQueryable extends BaseQuery {
    * @param {string} fieldUid - field uid to exclude
    * @returns {EntryQueryable} - returns EntryQueryable object for chaining method calls
    */
-  except(fieldUid: string): EntryQueryable {
-    this._queryParams['except[BASE][]'] = fieldUid;
+  except(fieldUid: string|string[]): EntryQueryable {
```
Comment (sunil-lakshman) [inline src/lib/entry-queryable.ts:45]: @vkalta Can you add the testcases for this implementation?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@contentstack/delivery-sdk",
-  "version": "4.4.0",
+  "version": "4.4.1",
```
Comment (sunil-lakshman) [inline package.json:3]: @vkalta Add version in changelog file.

**Comment by nadeem-cs** [comment]
That's one way to add the params, however, can you try adding array for the same key or using URLSearchParams? 
Here's a stackoverflow reference - https://stackoverflow.com/a/43208627

**Comment by vikramkalta** [comment]
> That's one way to add the params, however, can you try adding array for the same key or using URLSearchParams? Here's a stackoverflow reference - https://stackoverflow.com/a/43208627

Hi @nadeem-cs , I could have done that but the API calls are handled by contentstack/core library.

**Review by nadeem-cs — DISMISSED** [review]

File: src/lib/entry-queryable.ts (hunk 1 above)
Comment (vkalta) [reply src/lib/entry-queryable.ts:18]: Test cases added

File: src/lib/base-query.ts [hunk 4]
```diff
@@ -209,15 +209,14 @@ export class BaseQuery extends Pagination {
       requestParams = { ...this._queryParams, query: { ...this._parameters } };
     }
 
-    const getRequestOptions: any = { params: requestParams };
+    const getRequestOptions: any = requestParams;
```
Comment (nadeem-cs) [inline src/lib/base-query.ts:212]: @vkalta We have changed the third param of getData function. It now takes an object having params and headers both. Kindly revert this line.

**Review by nadeem-cs — CHANGES_REQUESTED** [review]

File: src/lib/base-query.ts (hunk 4 above)
Comment (vikramkalta) [reply src/lib/base-query.ts]: I have reverted the changes @nadeem-cs

Comment (vkalta) [reply src/lib/base-query.ts]: I have reverted the changes @nadeem-cs

**Review by nadeem-cs — APPROVED** [review]

**Pull request merged**